tests/e2e/specs/shopper/mini-cart.test.js

```diff
@@ -94,4 +123,89 @@ describe( 'Shopper → Mini Cart', () => {
 			);
 		} );
 	} );
+
+	describe( 'Filled mini cart', () => {
+		beforeAll( async () => {
+			await shopper.emptyCart();
+		} );
+
+		afterEach( async () => {
+			await shopper.emptyCart();
+		} );
+
+		it( 'The Mini Cart title shows correct amount', async () => {
+			await page.click(
+				'.wc-block-grid__product:first-child .add_to_cart_button'
+			);
+
+			await expect( page ).toMatchElement( '.wc-block-mini-cart__title', {
+				text: 'Your cart (1 item)',
+			} );
+
+			await page.mouse.click( 50, 200 );
+
+			await page.click(
+				'.wc-block-grid__product:last-child .add_to_cart_button'
+			);
+
+			await expect( page ).toMatchElement( '.wc-block-mini-cart__title', {
+				text: 'Your cart (2 items)',
+			} );
+		} );
+
+		it( 'The Mini Cart products table show added products', async () => {
+			const products = await page.$$(
+				'.wc-block-all-products .wc-block-grid__product'
+			);
+			if ( products.length === 0 ) {
+				return;
+			}
+
+			// Get a random product to ensure the test isn't false positive.
+			const product =
+				products[ Math.floor( Math.random() * products.length ) ];
+			const productTitle = await page.evaluate( ( el ) => {
+				return el.querySelector( '.wc-block-components-product-name' )
+					.textContent;
+			}, product );
+			await page.evaluate( ( el ) => {
+				el.querySelector( '.add_to_cart_button' ).click();
+			}, product );
```

Comment:
is there a reason for why we sometimes use it like so and other times just via `await page.click( '.add_to_cart_button' );` ?

Reply:
got it, we are clicking on the product related addToCart 😅